.github/workflows/lint.yaml

```diff
@@ -0,0 +1,50 @@
+name: lint
+
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+    branches:
+      - main
+      - feature/*
+  push:
+    branches:
+      - main
+      - feature/*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ inputs.platform }} @ ${{ github.event.label.name || github.event.pull_request.number || github.sha }} @ ${{ github.event.label.name && github.event.pull_request.number || github.event.action }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  lint:
+    name: Pre-Commit
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Install pre-commit
+        run: |
+          sudo apt update
+          sudo apt install python3 gn pipx
+          pipx install pre-commit
+      - name: Checkout files
+        uses: actions/checkout@v4
```

Comment:
Is this the same version we used for Cobalt branch ? We had a forked one due to unfixed bugs with subtrees. It may be fine to use the latest here but we may also get surprised again.

Reply:
It's the same as the ones used for Cobalt in main.yaml.